tests/integration/userns.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	setup_busybox
+
+	# Prepare source folders for bind mount
+	mkdir -p source-{accessible,inaccessible-1,inaccessible-2}/dir
+	touch source-{accessible,inaccessible-1,inaccessible-2}/dir/foo.txt
+
+	# Permissions only to the owner, it is inaccessible to group/others
+	chmod 700 source-inaccessible-{1,2}
+
+	mkdir -p rootfs/{proc,sys,tmp}
+	mkdir -p rootfs/tmp/mount-{1,2}
+
+	# We need to give permissions for others so the uid inside the userns
+	# can mount the rootfs on itself. Otherwise the rootfs mount will fail.
+	chmod 755 "$ROOT"
+
+	if [ "$ROOTLESS" -eq 0 ]; then
+		update_config ' .linux.namespaces += [{"type": "user"}]
+			| .linux.uidMappings += [{"hostID": 100000, "containerID": 0, "size": 65534}]
+			| .linux.gidMappings += [{"hostID": 100000, "containerID": 0, "size": 65534}] '
+	fi
+}
+
+function teardown() {
+	teardown_bundle
+}
+
+@test "userns with simple mount" {
+	update_config ' .process.args += ["-c", "stat /tmp/mount-1/foo.txt"]
+		| .mounts += [{"source": "source-accessible/dir", "destination": "/tmp/mount-1", "options": ["bind"]}] '
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+}
```

Comment:
Doesn't seem to work on Fedora 35
https://github.com/opencontainers/runc/pull/3258

Reply:
I mentioned it in several places already, but just in case someone is looking here, the PR fixing this is: https://github.com/opencontainers/runc/pull/3260